src/transformers/modeling_utils.py

```diff
@@ -511,7 +511,10 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike], is_quantized: bool
                 f"The safetensors archive passed at {checkpoint_file} does not contain the valid metadata. Make sure "
                 "you save your model with the `save_pretrained` method."
             )
-        return safe_load_file(checkpoint_file)
+        state_dict = safe_load_file(checkpoint_file)
+        del metadata["format"]
+        state_dict.update(metadata)
+        return state_dict
```

Comment:
I have to merge the metadata with the state dict, which is ugly. Maybe there is a better way to do that ?